src/Turbo/Mercure/Broadcaster.php

```diff
@@ -0,0 +1,160 @@
+        $this->publisher = $publisher;
+        $this->expressionLanguage = $expressionLanguage ?? new ExpressionLanguage();
+        $this->entityNamespace = $entityNamespace;
+    }
+
+    public function broadcast(object $entity, string $action): void
+    {
+        if (!$attribute = (new \ReflectionClass($entity))->getAttributes(Broadcast::class)[0] ?? null) {
+            return;
+        }
+
+        /**
+         * @var Broadcast $broadcast
+         */
+        $broadcast = $attribute->newInstance();
+        $options = $this->normalizeOptions($entity, $action, $broadcast->options);
+
+        if (isset($options['transports']) && !\in_array($this->name, $options['transports'], false)) {
+            return;
+        }
+
+        // What must we do if the template or the block doesn't exist? Throwing for now.
+        $data = $this->twig->load($options['template'])->renderBlock($action, ['entity' => $entity, 'action' => $action, 'options' => $options]);
+
+        $update = new Update(
+            $options['topics'],
+            $data,
+            $options['private'] ?? false,
+            $options['id'] ?? null,
+            $options['type'] ?? null,
+            $options['retry'] ?? null
+        );
+
+        $this->publisher->publish($update);
+    }
+
+    /**
+     * @param mixed[] $options
+     *
+     * @return mixed[]
+     */
+    private function normalizeOptions(object $entity, string $action, array $options): array
+    {
+        if (isset($options['transports'])) {
+            $options['transports'] = (array) $options['transports'];
+        }
+
+        if (\is_string($options[0] ?? null)) {
+            if (null === $this->expressionLanguage) {
+                throw new \RuntimeException('The Expression Language component is not installed. Try running "composer require symfony/expression-language".');
+            }
+
+            $options = $this->expressionLanguage->evaluate($options[0], ['entity' => $entity, 'action' => $action]);
+        }
+
+        $entityClass = \get_class($entity);
+
+        if ($extraKeys = array_diff(array_keys($options), self::OPTIONS)) {
+            throw new \InvalidArgumentException(sprintf('Unknown broadcast options "%s" on class "%s". Valid options are: "%s"', implode('", "', $extraKeys), $entityClass, implode('", "', self::OPTIONS)));
+        }
+
+        $options['topics'] = (array) ($options['topics'] ?? sprintf(self::TOPIC_PATTERN, rawurlencode($entityClass), rawurlencode($this->propertyAccessor->getValue($entity, 'id'))));
+        if (isset($options['template'])) {
+            return $options;
+        }
+
+        $dir = $entityClass;
+        if ($this->entityNamespace && 0 === strpos($entityClass, $this->entityNamespace)) {
+            $dir = substr($entityClass, \strlen($this->entityNamespace));
+        }
+
+        $options['template'] = sprintf('broadcast/%s.stream.html.twig', str_replace('\\', '/', $dir));
```

Comment:
Does this approach work with third party broadcasts?

Reply:
Yes you can plug your own transport (it's documented at the end of the readme file).